src/asm/lexer.c

```diff
@@ -1174,6 +1174,95 @@ static int readIdentifier(char firstChar)
 	return tokenType;
 }
 
+/* Function to read identifiers & keywords for macros within macros */
+
+static bool startsMacroWithinMacro(int c)
+{
+	return (c <= 'Z' && c >= 'A') || (c <= 'z' && c >= 'a') || c == '_' || c == '\\';
+}
+
+static int readMacroWithinMacro(char firstChar)
```

Comment:
I don't like this name, because it seems to imply that the function reads the entire macro, when it only checks for a macro boundary. Maybe `readMacroBoundary`? Though I'm not satisfied with that one...